### PR TITLE
Added 'post_binding_setup' for each binding available

### PR DESCRIPTION
**Pull request by mikkac**
Refers #2581

### CODE REVIEW

**Comment by mikkac** [comment]
AFAIK there already is `post_markdown_setup` defined in `src/mlpack/bindings/markdown/CMakeLists.txt` and called in `src/mlpack/CMakeLists.txt`. Right?

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,38 +29,6 @@ else ()
       "Compile shared libraries (if OFF, static libraries are compiled)." ON)
 endif()
 
-# Detect whether the user passed BUILD_PYTHON_BINDINGS in order to determine if
```
Comment (Yashwants19) [inline CMakeLists.txt:32]: Hi @mikkac, these configuration are at right place. You have to target the setup present in `src/mlpack/CMakeLists.txt` i.e.
```Cmake
if (BUILDING_PYTHON_BINDINGS)
...
endif()

if (BUILDING_JULIA_BINDINGS)
...
endif()

if (BUILDING_GO_BINDINGS)
...
endif()
```

Comment (rcurtin) [reply CMakeLists.txt:32]: Right, exactly.  Sorry @mikkac if the issue was unclear---the part to refactor is not this bit, but the stuff in `src/mlpack/CMakeLists.txt` that's kind of like `post_markdown_setup()`.  In fact you could see `post_markdown_setup()` as "factored out", and so the idea is to take the stuff that's in `src/mlpack/CMakeLists.txt` at the end and related to each individual binding type, and then just put that in functions with exactly the name that you've got here.  Let me know if I can clarify that. :+1:  And thanks for taking the time to work on this! :rocket:

Comment (mikkac) [reply CMakeLists.txt:32]: Hi, sure- I could misunderstand the issue's description, no problem :) I will take those tips into consideration and try to work something out in free time- sorry, currently I'm very busy with other projects. Stay tuned!

Comment (rcurtin) [reply CMakeLists.txt:32]: Sounds good---no worries.  Good luck with the other projects! :+1:

**Comment by mikkac** [comment]
Hi, since I misunderstood the task and the code I've written is all wrong, I've opened new pull request #2608

**Pull request closed without merging**